File: CS/MeffedNavigationService.Infrastructure/DataModel/MenuDataItem.cs
using System;
using System.Collections.ObjectModel;
using System.Linq;
using DevExpress.Xpf.Mvvm;

namespace MeffedNavigationService.Infrastructure.DataModel {
    public class MenuDataItem : ViewModelBase {
        private int _uniqueID;
        private string _groupHeader;
        private string _title;
        private bool _isFlowBreak;
        private MeffedNavigationService.Infrastructure.Enums.ModuleTypes _moduleType;
        static int idCount = 0;
        /// <summary>
        /// Initializes a new instance of the MenuDataItem class.
        /// </summary>
        /// <param name="uniqueId">The unique id.</param>
        /// <param name="title">The title.</param>
        /// <param name="groupHeader">The group header.</param>
        public MenuDataItem(string title, MeffedNavigationService.Infrastructure.Enums.ModuleTypes moduleType, string groupHeader, bool isFlowBreak) {
            ModuleType = moduleType;
            IsFlowBreak = isFlowBreak;
            UniqueID = idCount++;
            GroupHeader = groupHeader;
            Title = title;
        }
        /// <summary>
        /// Initializes a new instance of the MenuDataItem class.
        /// </summary>
        /// <param name="uniqueId">The unique id.</param>
        /// <param name="title">The title.</param>
        public MenuDataItem(string title, MeffedNavigationService.Infrastructure.Enums.ModuleTypes moduleType)
            : this(title, moduleType, String.Empty, false) {
        }
        public int UniqueID {
            get { return _uniqueID; }
            private set { this.SetProperty(ref this._uniqueID, value, "UniqueID"); }
        }
        public MeffedNavigationService.Infrastructure.Enums.ModuleTypes ModuleType {
            get { return _moduleType; }
            set { this.SetProperty(ref this._moduleType, value, "ModuleType"); }
        }
        public bool IsFlowBreak {
            get { return _isFlowBreak; }
            set { this.SetProperty(ref this._isFlowBreak, value, "IsFlowBreak"); }
        }
        public string GroupHeader {
            get { return _groupHeader; }
            set { this.SetProperty(ref this._groupHeader, value, "GroupHeader"); }
        }
        public string Title {
            get { return _title; }
            set { this.SetProperty(ref this._title, value, "Title"); }
        }
        public override string ToString() {
            return this.Title;
        }
    }
    public sealed class SampleMenuDataSource {
        public static SampleMenuDataSource Instance {
            get { return _sampleMenuDataSource; }
        }
        private static readonly SampleMenuDataSource _sampleMenuDataSource = new SampleMenuDataSource();
        private ObservableCollection<MenuDataItem> _items;
        public ObservableCollection<MenuDataItem> Items {
            get { return Instance._items; }
        }
        public static MenuDataItem GetItem(int uniqueId) {
            var matches = SampleMenuDataSource.Instance.Items.Where((item) => item.UniqueID.Equals(uniqueId));
            if(matches.Count() == 1) return matches.First();
            return null;
        }
        /// <summary>
        /// Initializes a new instance of the SampleMenuDataSource class.
        /// </summary>
        public SampleMenuDataSource() {
            _items = new ObservableCollection<MenuDataItem>();
            _items.Add(new MenuDataItem("Front Desk", MeffedNavigationService.Infrastructure.Enums.ModuleTypes.ViewFrontDesk, "Guest Services", true));
            _items.Add(new MenuDataItem("Management", MeffedNavigationService.Infrastructure.Enums.ModuleTypes.ViewManagement));
        }
    }
}

File: VB/MeffedNavigationService.Infrastructure/DataModel/MenuDataItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.ObjectModel
Imports System.Linq
Imports DevExpress.Xpf.Mvvm

Namespace MeffedNavigationService.Infrastructure.DataModel
	Public Class MenuDataItem
		Inherits ViewModelBase
		Private _uniqueID As Integer
		Private _groupHeader As String
		Private _title As String
		Private _isFlowBreak As Boolean
		Private _moduleType As MeffedNavigationService.Infrastructure.Enums.ModuleTypes
		Private Shared idCount As Integer = 0
		''' <summary>
		''' Initializes a new instance of the MenuDataItem class.
		''' </summary>
		''' <param name="uniqueId">The unique id.</param>
		''' <param name="title">The title.</param>
		''' <param name="groupHeader">The group header.</param>
		Public Sub New(ByVal title As String, ByVal moduleType As MeffedNavigationService.Infrastructure.Enums.ModuleTypes, ByVal groupHeader As String, ByVal isFlowBreak As Boolean)
			Me.ModuleType = moduleType
			Me.IsFlowBreak = isFlowBreak
			UniqueID = idCount
			idCount += 1
			Me.GroupHeader = groupHeader
			Me.Title = title
		End Sub
		''' <summary>
		''' Initializes a new instance of the MenuDataItem class.
		''' </summary>
		''' <param name="uniqueId">The unique id.</param>
		''' <param name="title">The title.</param>
		Public Sub New(ByVal title As String, ByVal moduleType As MeffedNavigationService.Infrastructure.Enums.ModuleTypes)
			Me.New(title, moduleType, String.Empty, False)
		End Sub
		Public Property UniqueID() As Integer
			Get
				Return _uniqueID
			End Get
			Private Set(ByVal value As Integer)
				Me.SetProperty(Me._uniqueID, value, "UniqueID")
			End Set
		End Property
		Public Property ModuleType() As MeffedNavigationService.Infrastructure.Enums.ModuleTypes
			Get
				Return _moduleType
			End Get
			Set(ByVal value As MeffedNavigationService.Infrastructure.Enums.ModuleTypes)
				Me.SetProperty(Me._moduleType, value, "ModuleType")
			End Set
		End Property
		Public Property IsFlowBreak() As Boolean
			Get
				Return _isFlowBreak
			End Get
			Set(ByVal value As Boolean)
				Me.SetProperty(Me._isFlowBreak, value, "IsFlowBreak")
			End Set
		End Property
		Public Property GroupHeader() As String
			Get
				Return _groupHeader
			End Get
			Set(ByVal value As String)
				Me.SetProperty(Me._groupHeader, value, "GroupHeader")
			End Set
		End Property
		Public Property Title() As String
			Get
				Return _title
			End Get
			Set(ByVal value As String)
				Me.SetProperty(Me._title, value, "Title")
			End Set
		End Property
		Public Overrides Function ToString() As String
			Return Me.Title
		End Function
	End Class
	Public NotInheritable Class SampleMenuDataSource
		Public Shared ReadOnly Property Instance() As SampleMenuDataSource
			Get
				Return _sampleMenuDataSource
			End Get
		End Property
		Private Shared ReadOnly _sampleMenuDataSource As New SampleMenuDataSource()
		Private _items As ObservableCollection(Of MenuDataItem)
		Public ReadOnly Property Items() As ObservableCollection(Of MenuDataItem)
			Get
				Return Instance._items
			End Get
		End Property
		Public Shared Function GetItem(ByVal uniqueId As Integer) As MenuDataItem
			Dim matches = SampleMenuDataSource.Instance.Items.Where(Function(item) item.UniqueID.Equals(uniqueId))
			If matches.Count() = 1 Then
				Return matches.First()
			End If
			Return Nothing
		End Function
		''' <summary>
		''' Initializes a new instance of the SampleMenuDataSource class.
		''' </summary>
		Public Sub New()
			_items = New ObservableCollection(Of MenuDataItem)()
			_items.Add(New MenuDataItem("Front Desk", MeffedNavigationService.Infrastructure.Enums.ModuleTypes.ViewFrontDesk, "Guest Services", True))
			_items.Add(New MenuDataItem("Management", MeffedNavigationService.Infrastructure.Enums.ModuleTypes.ViewManagement))
		End Sub
	End Class
End Namespace